cogs/challenges.py

```diff
@@ -54,17 +52,24 @@ async def on_message(self, message):  # Submitted role.
 
             if submitted not in message.author.roles:
                 await message.delete()
-                if message.content.count("```") != 2:
-                    msg = f"{message.author.mention} make sure to submit in a code " \
-                          f"block and only include the code required for the challenge!" \
-                          f"\nUse `tim.tag discord code` for more information!"
+                attach = message.attachments and message.attachments[0]
+                
+                if not attach:
+                    msg = f"{message.author.mention} make sure to __upload a " \
+                          f"python file__ that only includes the code required " \
+                          f"for the challenge!"
+                    return await message.channel.send(msg, delete_after=10.0)
+                
+                content = "```py\n" + (await attach.read()).decode("u8").replace("`", "\u200b`") + "```"
+                if len(content) > 2040:
+                    msg = f"{message.author.mention} attachment can't be __more" \
+                          f" than 2040 characters__."
```

Comment:
shouldn't this be 2031: `content - "```py\n```"`. Or change `if len(content) > 2040` to `if len(content) > 2049` :ugundu:

Reply:
`if len(content) > 2049` ? But the description limit is `2048`